DorisProject/src/app/pages/stock/add-stock/add-stock.component.ts

```diff
@@ -65,46 +76,86 @@ export class AddStockComponent implements OnInit {
     dialogRef.afterClosed().subscribe(result => {
+      this.responseModal = result;
+      switch (this.responseModal) {
+        case 'save':
+          this.save();
+          break;
+
+      }
     });
   }
 
 
   get errorMesageDescription() {
-        return this.formGroupProduct.get('description').invalid && this.formGroupProduct.get('description').touched;
+    return this.formGroupProduct.get('description').invalid && this.formGroupProduct.get('description').touched;
   }
+
   get errorMesageBrand() {
-        return this.formGroupProduct.get('brand').invalid && this.formGroupProduct.get('brand').touched;
+    return this.formGroupProduct.get('brand').invalid && this.formGroupProduct.get('brand').touched;
   }
+
   get errorMesageStock() {
-        return this.formGroupProduct.get('stock').invalid && this.formGroupProduct.get('stock').touched;
+    return this.formGroupProduct.get('stock').invalid && this.formGroupProduct.get('stock').touched;
   }
+
   get errorMesageIdCategory() {
-        return this.formGroupProduct.get('idCategory').invalid && this.formGroupProduct.get('idCategory').touched;
+    return this.formGroupProduct.get('idCategory').invalid && this.formGroupProduct.get('idCategory').touched;
   }
+
   get errorMesageActive() {
-        return this.formGroupProduct.get('active').invalid && this.formGroupProduct.get('active').touched;
+    return this.formGroupProduct.get('active').invalid && this.formGroupProduct.get('active').touched;
   }
+
   get errorMesageSku() {
-        return this.formGroupProduct.get('sku').invalid && this.formGroupProduct.get('sku').touched;
+    return this.formGroupProduct.get('sku').invalid && this.formGroupProduct.get('sku').touched;
   }
+
   get errorMesageName() {
-        return this.formGroupProduct.get('name').invalid && this.formGroupProduct.get('name').touched;
+    return this.formGroupProduct.get('name').invalid && this.formGroupProduct.get('name').touched;
   }
 
 
-  save() {
+  loadUpdate() {
+    this.addstockservice.getProductBysku(this.product.Sku).subscribe((response) => {
+        console.log('request al servicio getProductSku ');
+
+        if (response != null) {
+          console.log('getProductSku Response: ', response);
+          this.product = response;
+          this.formGroupProduct.setValue({
+            id: this.product.id,
+            name: this.product.name,
+            idCategory: this.product.idcategory,
+            brand: this.product.brand,
+            description: this.product.description,
+            stock: this.product.stock,
+            active: this.product.active,
+            sku: this.product.sku,
+
+          });
+        }
+
+      }, err => {
+        console.log('error: ', err);
+      }
+    );
+  }
+
+  update() {
     const data: Product = this.formGroupProduct.getRawValue();
 
     console.log(data);
-    this.addstockservice.save(data).subscribe(response => {
+    this.addstockservice.update(data).subscribe(res => {
```

Comment:
Codacy has a fix for the issue: [Replace `res` with `⏎······(res)`](https://app.codacy.com/gh/SaraArenasV/DorisProjectFrontEnd/pullRequest?prid=9258681)
```suggestion
    this.addstockservice.update(data).subscribe(
```